docs/spec.md

```diff
@@ -3273,25 +3306,31 @@ the future ([#487](https://github.com/openxla/stablehlo/issues/487)).
 
 #### Inputs
 
-| Label | Name          | Type                                         | Constraints |
-|-------|---------------|----------------------------------------------|-------------|
-| (I1)  | `inputs`      | variadic number of tensors                   | (C1-C4)     |
-| (I2)  | `dimensions`  | 1-dimensional tensor constant of type `si64` | (C3)        |
-| (I3)  | `computation` | function                                     | (C4)        |
+| Label | Name          | Type                                                       | Constraints |
+|-------|---------------|------------------------------------------------------------|-------------|
+| (I1)  | `inputs`      | variadic number of tensors or per-tensor quantized tensors | (C1-C6)     |
+| (I2)  | `dimensions`  | 1-dimensional tensor constant of type `si64`               | (C3)        |
+| (I3)  | `computation` | function                                                   | (C4), (C6)  |
 
 #### Outputs
 
-| Name     | Type   | Constraints |
-|----------|--------|-------------|
-| `result` | tensor | (C1), (C4)  |
+| Name     | Type                                  | Constraints      |
+|----------|---------------------------------------|------------------|
+| `result` | tensor or per-tensor quantized tensor | (C1), (C4), (C6) |
 
 #### Constraints
 
 * (C1) `shape(inputs...) = shape(result)`.
 * (C2) `0 < size(inputs) = N`.
 * (C3) `dimensions = range(rank(inputs[0]))`.
-* (C4) `computation` has type `(tensor<E0>, ..., tensor<EN-1>) -> tensor<E'>`
-  where `Ei = element_type(inputs[i])` and `E' = element_type(result)`.
+* If the operation uses non-quantized tensors:
+  * (C4) `computation` has type `(tensor<E0>, ..., tensor<EN-1>) -> tensor<E'>`
+    where `Ei = element_type(inputs[i])` and `E' = element_type(result)`.
+* If the operation uses quantized tensors:
+  * (C5) is_quantized_tensor(inputs[i]) for all `i in range(size(inputs))`.
```

Comment:
Do we really need C5 here? Given that MapOp is just a passthrough op, I think the individual element types shouldn't matter to it? This way, we'll also be able to merge C4 and C6.

Reply:
>> Do we really need C5 here? Given that MapOp is just a passthrough op, I think the individual element types shouldn't matter to it? 
agreed.

>> This way, we'll also be able to merge C4 and C6.
We still need different member_name function like `element_type` and `quantized_element_type` to cover for non-quantized and quantized cases. right? `baseline_element_type` may not work here as the we want the block arguments to match exactly the input types and block return to match the output type.

Reply:
1) "We still need different member_name function like element_type and quantized_element_type to cover for TensorType and QuantizedTensorType cases. right?". Sure. We can do something like `is_quantized(inputs[i]) ? quantized_element_type(inputs[i]) : element_type(inputs[i])`.

2) "With that we can define define element_type as a function on types such that...". I think it's useful to keep the two separate, to ensure that handling of a quantized type is an opt in (which is in line with the design principles that we've been using so far, e.g. by splitting tensor type and quantized tensor type).

3) "NonValueType ::= ElementType | FunctionType | StringType". Agreed, this should include QuantizedElementType as well.

Reply:
Done.
Will address (3) in a separate PR.